OpenRA.Mods.Common/Traits/Burns.cs

```diff
@@ -9,50 +9,62 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using OpenRA.Graphics;
 using OpenRA.Mods.Common.Traits.Render;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("This actor will play a fire overlay animation over its body and take damage over time.")]
-	class BurnsInfo : ITraitInfo, Requires<RenderSpritesInfo>
+	class BurnsInfo : PausableConditionalTraitInfo, Requires<RenderSpritesInfo>
 	{
 		public readonly string Image = "fire";
 		[SequenceReference("Image")] public readonly string Anim = "1";
 
 		[Desc("Custom palette name")]
 		[PaletteReference("IsPlayerPalette")] public readonly string Palette = null;
 
-		[Desc("Custom palette is a player palette BaseName")]
+		[Desc("Custom palette is a player palette BaseName.")]
 		public readonly bool IsPlayerPalette = false;
 
+		[Desc("Amount of damage received per Interval ticks.")]
 		public readonly int Damage = 1;
+
+		[Desc("Delay between receiving damage.")]
 		public readonly int Interval = 8;
 
-		public object Create(ActorInitializer init) { return new Burns(init.Self, this); }
+		[Desc("Apply the damage using these damagetypes.")]
+		public readonly HashSet<string> DamageTypes = new HashSet<string>();
+
+		public override object Create(ActorInitializer init) { return new Burns(init.Self, this); }
 	}
 
-	class Burns : ITick, ISync
+	class Burns : PausableConditionalTrait<BurnsInfo>, ITick, ISync
 	{
 		readonly BurnsInfo info;
 		[Sync] int ticks;
 
 		public Burns(Actor self, BurnsInfo info)
+			: base(info)
 		{
 			this.info = info;
 
-			var anim = new Animation(self.World, info.Image, () => 0);
+			var anim = new Animation(self.World, info.Image, () => 0, () => IsTraitPaused);
 			anim.IsDecoration = true;
+			var animWithOffset = new AnimationWithOffset(anim, () => WVec.Zero, () => IsTraitDisabled, p => RenderUtils.ZOffsetFromCenter(self, p, 1));
+			self.Trait<RenderSprites>().Add(animWithOffset, info.Palette, info.IsPlayerPalette);
 			anim.PlayRepeating(info.Anim);
```

Comment:
Shouldn't this be `animWithOffset` then? (note: I haven't checked, I'm just asking)

Reply:
I believe this also confused me, but I checked against existing code on how to use this.